tests/layering/helpers_test.go

```diff
@@ -0,0 +1,611 @@
+func assertNotInDerivedImage(t *testing.T, cs *framework.ClientSet, node *corev1.Node) {
+	status, err := getRPMOStreeStatus(t, cs, node)
+	require.Nil(t, err)
+	checkUsingImage(t, cs, node, false, status)
+}
+
+// Performs the check that we're either in the derived image or not.
+func checkUsingImage(t *testing.T, cs *framework.ClientSet, node *corev1.Node, usingImage bool, status *Status) {
+	// These files are placed on the node by the derived container build process.
+	expectedFiles := []string{
+		"/usr/lib64/libpixman-1.so.0",
+		"/etc/systemd/system/hello-world.service",
+		helloWorldPath,
+	}
+
+	for _, deployment := range status.Deployments {
+		if deployment.Booted {
+			if usingImage {
+				t.Logf("we expect that we're using the newly derived image")
+				// Check that our container image is as expected
+				assert.Equal(t, imageURL, deployment.ContainerImageReference)
+				// Check that we have the expected files on the node
+				assertNodeHasFiles(t, cs, node, expectedFiles)
+				// Run the hello world program that we built and placed on the node
+				result, err := ExecCmdOnNode(cs, node, []string{helloWorldPath})
+				t.Logf("Running hello world command: \n%s", result.String())
+				assert.Nil(t, err)
+			} else {
+				t.Logf("we expect that we're using the non-derived image")
+				assert.NotEqual(t, imageURL, deployment.ContainerImageReference)
+				assertNodeNotHasFiles(t, cs, node, expectedFiles)
+			}
+		}
+	}
+}
+
+// Asserts that a node has files in the expected places.
+func assertNodeHasFiles(t *testing.T, cs *framework.ClientSet, node *corev1.Node, files []string) {
+	for _, file := range files {
+		out, err := ExecCmdOnNode(cs, node, []string{"stat", file})
+		require.Nil(t, err, "expected to find %s on node %s: %s", file, node.Name, out.String())
+		t.Logf("found %s on node %s (this was as expected)", file, node.Name)
+	}
+}
+
+// Asserts that a node does not have files in the expected places.
+func assertNodeNotHasFiles(t *testing.T, cs *framework.ClientSet, node *corev1.Node, files []string) {
+	for _, file := range files {
+		out, err := ExecCmdOnNode(cs, node, []string{"stat", file})
+		require.NotNil(t, err, "expected not to find %s on node %s:\n%s", file, node.Name, out.String())
+		t.Logf("did not find %s on node %s (this was as expected)", file, node.Name)
+	}
+}
+
+// Determines if a node is ready.
+func isNodeReady(node *corev1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status == "True" {
+			return true
+		}
+	}
+
+	return false
+}
+
+// RebootAndWait reboots a node, waits until it's status is ready.
+func rebootAndWait(t *testing.T, cs *framework.ClientSet, node *corev1.Node) error {
+	t.Logf("rebooting %s", node.Name)
+
+	// Get an updated node object since this one may potentially be out-of-date
+	updatedNode, err := cs.Nodes().Get(context.TODO(), node.ObjectMeta.Name, metav1.GetOptions{})
```

Comment:
```suggestion
	updatedNode, err := cs.Nodes().Get(context.TODO(), node.Name, metav1.GetOptions{})
```
You shouldn't need ObectMeta to get name for node of corev1.Node type for example in our repo: https://github.com/openshift/machine-config-operator/blob/9c6c2bfd7ed498bfbc296d530d1839bd6a177b0b/pkg/daemon/node.go#L24

Reply:
Yeah, I'll go through and fix all of those. I'm not sure why I did that ¯\_(ツ)_/¯